packages/ui-components/src/components/windowedlist.ts

```diff
@@ -267,32 +267,36 @@ export abstract class WindowedListModel implements WindowedList.IModel {
    *
    * @param index Item index
    * @param align Where to align the item in the viewport
+   * @param margin In 'smart' mode the viewport proportion to add
    * @returns The needed scroll offset
    */
   getOffsetForIndexAndAlignment(
     index: number,
-    align: WindowedList.ScrollToAlign = 'auto'
+    align: WindowedList.ScrollToAlign = 'auto',
+    margin: number = 0.25
   ): number {
+    const margin_ =
```

Comment:
The name `margin_` seems a bit confusing here; should we have a more descriptive name?